### ISSUE TITLE
Depreciate MATEmbedding layer

### ISSUE THREAD

**Issue by arunppsg** [deepchem/deepchem#3147]
MATEmbedding kayer is used to generate an embedding of output node features in `torch_models.mat.MAT`  model. The layer wraps `nn.Linear` and `nn.Dropout` layers. It does not add any other functionality. To replace it, we can directly add a Linear and Dropout and linear layer in `torch_models.mat.MAT` where it is being used. 
I would like to suggest depreciating it for two reasons:
- with less number of choices, users will be able to decide better on what layers to use.
- lesser lines of code, easier to maintain

Before depreciating, we should consider how depreciating it will affect reloading of existing models MAT (Molecular Attention Transformer) models.
